platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,206 @@
+
+    private final ConfiguratorFactory configuratorFactory;
+
+    private final ConfigActions configActions;
+
+    private Map<String, File> pidFileMap;
+
+    public ConfigurationInstaller(ConfigurationAdmin configurationAdmin,
+            ConfiguratorFactory configuratorFactory, ConfigActions configActions) {
+        this.configurationAdmin = configurationAdmin;
+        this.pidFileMap = new ConcurrentHashMap<>();
+
+        this.configuratorFactory = configuratorFactory;
+        this.configActions = configActions;
+    }
+
+    /**
+     * @return a read-only map of configuration pids to felix file install properties.
+     */
+    public Map<String, File> getPidFileMap() {
+        return Collections.unmodifiableMap(pidFileMap);
+    }
+
+    public void init() throws IOException, InvalidSyntaxException {
+        Configuration[] configs = configurationAdmin.listConfigurations(null);
+        if (configs == null) {
+            return;
+        }
+        this.pidFileMap = Arrays.stream(configs)
+                .map(FelixConfig::new)
+                .filter(config -> config.getFile() != null)
+                .collect(Collectors.toConcurrentMap(FelixConfig::getPid, FelixConfig::getFile));
+    }
+
+    @Override
+    public void configurationEvent(ConfigurationEvent event) {
+        String pid = event.getPid();
+        switch (event.getType()) {
+        case CM_UPDATED:
+            LOGGER.debug("Handling update for pid {}", pid);
+            handleUpdate(event);
+            break;
+        case CM_DELETED:
+            LOGGER.debug("Handling delete for pid {}", pid);
+            handleDelete(event);
+            break;
+        default:
+            LOGGER.debug("Unknown configuration event type, taking no action for pid {}", pid);
+        }
+    }
+
+    /**
+     * Sync the config files in etc when an update occurs.
+     *
+     * @param event the configuration event for which an update is needed
+     */
+    private void handleUpdate(ConfigurationEvent event) {
+        Configuration configuration = getConfiguration(event);
+        if (configuration == null) {
+            LOGGER.debug("Configuration was null for pid {}", event.getPid());
+            return;
+        }
+
+        FelixConfig felixConfig = new FelixConfig(configuration);
+        String pid = felixConfig.getPid();
+        File felixFileNew = felixConfig.getFile();
+        File felixFileOriginal = pidFileMap.get(pid);
+
+        if (felixFileOriginal == null && felixFileNew == null) {
+            LOGGER.debug("Was not tracked and will not track pid {}", pid);
+            return;
+        }
+
+        boolean filePropChanged = felixConfig.filePropChanged(felixFileOriginal);
+
+        if (filePropChanged) {
+            if (felixFileOriginal != null) {
+                LOGGER.debug("Felix filename prop changed, deleting file for pid {}", pid);
+                performOperation(configActions.delete(felixFileOriginal.toPath()));
```

Comment:
@paouelle should we catch `RuntimeException` in the implemented `configurationEvent` method since retrieving the delete operation from `configActions` may throw an `IllegalArgumentException` if the path is outside DDF_HOME, among other criteria.